main/chains/index.js

```diff
@@ -83,47 +70,32 @@ class ChainConnection extends EventEmitter {
 
   _createBlockMonitor(provider) {
     const monitor = new BlockMonitor(provider)
-    const allowEip1559 = !legacyChains.includes(parseInt(this.chainId))
 
     monitor.on('data', async (block) => {
       log.debug(`Updating to block ${parseInt(block.number)} for chain ${parseInt(this.chainId)}`)
 
-      let feeMarket = null
-
-      const gasMonitor = new GasMonitor(provider)
-
-      if (allowEip1559 && 'baseFeePerGas' in block) {
-        try {
-          // only consider this an EIP-1559 block if fee market can be loaded
-          const feeHistory = await gasMonitor.getFeeHistory(10, [10])
-          feeMarket = this.gasCalculator.calculateGas(feeHistory)
+      if ('baseFeePerGas' in block && eip1559Allowed(this.chainId)) {
+        this.chainConfig.setHardforkByBlockNumber(block.number)
 
-          this.chainConfig.setHardforkByBlockNumber(block.number)
-
-          if (!this.chainConfig.gteHardfork(Hardfork.London)) {
-            // if baseFeePerGas is present in the block header, the hardfork
-            // must be at least London
-            this.chainConfig.setHardfork(Hardfork.London)
-          }
-        } catch (e) {
-          feeMarket = null
-          // log.error(`could not load EIP-1559 fee market for chain ${this.chainId}`, e)
+        if (!this.chainConfig.gteHardfork(Hardfork.London)) {
```

Comment:
I wonder if we can get rid of this kind of stuff now that we're so far past the London hardfork. @floating is there any reason to support gas calculation on eip-1559 chains for such old blocks?

Reply:
Leaving this in for now.